owlplug-client/src/main/java/com/owlplug/project/tasks/discovery/studioone/StudioOneAudioMixerPluginCollector.java

```diff
@@ -0,0 +1,172 @@
+/* OwlPlug
+ * Copyright (C) 2021 Arthur <dropsnorz@gmail.com>
+ *
+ * This file is part of OwlPlug.
+ *
+ * OwlPlug is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3
+ * as published by the Free Software Foundation.
+ *
+ * OwlPlug is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with OwlPlug.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.owlplug.project.tasks.discovery.studioone;
+
+import com.owlplug.plugin.model.PluginFormat;
+import com.owlplug.project.model.DawPlugin;
+import java.util.ArrayList;
+import java.util.List;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+public class StudioOneAudioMixerPluginCollector {
+
+  private final Logger log = LoggerFactory.getLogger(this.getClass());
+
+  private Document document;
+
+  public StudioOneAudioMixerPluginCollector(Document document) {
+    this.document = document;
+  }
+
+  public List<DawPlugin> collectPlugins() {
+
+    ArrayList<DawPlugin> plugins = new ArrayList<>();
+
+    XPath xpath = XPathFactory.newInstance().newXPath();
+    try {
+      Element insertsNode = findInsertsNode(document, xpath);
```

Comment:
It looks like this part is not working correctly. In my understanding, we are only retrieving the first `<Attributes x:id="Inserts">`, but in `audiomixer.xml`, we can find multiple *Inserts* in all `ChannelGroup.AudioInputChannel` and `ChannelGroup.AudioOutputChannel`.

To me, we should iterate over a list of *Inserts* element to find effects.